FordLabs/peoplemover
# Add typescript eslint config

## Issue
My editor was not understanding that enums were a thing because eslint did not have rules for typescript. I added that, which flagged a bunch of other warnings that I had to resolve.

## What was done
- [x] Added `plugin:@typescript-eslint/eslint-recommended` and `plugin:@typescript-eslint/recommended` to eslint's extend array
- [x] Upgraded a number of eslint packages 
- [x] Resolved all warnings and errors flagged after that 

## How to test
1. Ensure the app runs as it did before
2. Ensure all unit and cypress tests pass
